org.eclipse.gef.examples.ui.pde/src/org/eclipse/gef/examples/ui/pde/internal/wizards/ProjectUnzipperNewWizard.java

```diff
@@ -182,61 +176,36 @@ public boolean performFinish() {
 				@Override
 				public void execute(IProgressMonitor monitor) throws InterruptedException {
 					try {
-						monitor.beginTask(Messages.monitor_creatingProject, 120);
+						SubMonitor subMonitor = SubMonitor.convert(monitor, Messages.monitor_creatingProject, 150);
 
-						/*
-						 * Create the project folder
-						 */
 						IPath projectPath = wizardNewProjectCreationPage.getLocationPath();
 
 						for (int i = 0; i < nameFormats.length; i++) {
 							String projectName = MessageFormat.format(nameFormats[i],
-									new Object[] { wizardNewProjectCreationPage.getProjectName() });
-							String projectFolder = projectPath.toOSString() + File.separator + projectName;
-							File projectFolderFile = new File(projectFolder);
+									wizardNewProjectCreationPage.getProjectName());
 
 							IWorkspace workspace = ResourcesPlugin.getWorkspace();
 							IProject project = workspace.getRoot().getProject(projectName);
 
 							// If the project does not exist, we will create it
 							// and populate it.
 							if (!project.exists()) {
-								projectFolderFile.mkdirs();
-								monitor.worked(10);
-
-								/*
-								 * Copy plug-in project code
-								 */
-								extractProject(projectFolderFile, getProjectZipURL()[i],
-										new SubProgressMonitor(monitor, 100));
-
-								if (monitor.isCanceled()) {
-									throw new InterruptedException();
-								}
-
-								if (projectPath.equals(workspace.getRoot().getLocation())) {
-									project.create(monitor);
-								} else {
-									IProjectDescription desc = workspace.newProjectDescription(project.getName());
-									desc.setLocation(new Path(projectFolder));
-
-									project.create(desc, monitor);
-								}
+								createProject(subMonitor, projectPath, project, getProjectZipURL()[i]);
 							}
 
 							// Now, we ensure that the project is open.
-							project.open(monitor);
+							project.open(subMonitor.newChild(10));
 
 							renameProject(project, projectName);
 
 							// Add Java and PDE natures
 							IProjectDescription desc = workspace.newProjectDescription(project.getName());
 							desc.setNatureIds(
 									new String[] { ORG_ECLIPSE_PDE_PLUGIN_NATURE, ORG_ECLIPSE_JDT_CORE_JAVANATURE });
-							project.setDescription(desc, monitor);
+							project.setDescription(desc, subMonitor.newChild(10));
 
-							monitor.worked(10);
-							if (monitor.isCanceled()) {
+							subMonitor.worked(10);
```

Comment:
Now I wonder if it is really necessary to consume 10 ticks for "nothing", especially when 10 ticks were just consumed while setting the description.
```suggestion
```